src/mame/casio/pv1000.cpp

```diff
@@ -291,39 +304,50 @@ void pv1000_state::io_w(offs_t offset, uint8_t data)
 
 uint8_t pv1000_state::io_r(offs_t offset)
 {
+	/* Ports 0xF8-0xFB, 0xFE, and 0xFF are undriven, and pulled high by the
+	   resistors next to the Z80
+
+	   We can remove this bit when we adjust the range */
+
+	if (offset < 4 || offset > 5)
+		return 0xFF;
+
 	uint8_t data = m_io_regs[offset];
 
-//  logerror("io_r offset=%02x\n", offset);
 
-	switch (offset)
-	{
-	case 0x04:
-		/* Bit 1 = 1 => Data is available in port FD */
-		/* Bit 0 = 1 => Buffer at port FD is empty */
-		data = 0;
-		data = m_fd_buffer_flag & 1;
-		data |= m_fd_data ? 2 : 0;
-		m_fd_buffer_flag &= ~1;
-		break;
-	case 0x05:
-		static const char *const joynames[] = { "IN0", "IN1", "IN2", "IN3" };
+	static const char *const joynames[] = { "IN0", "IN1", "IN2", "IN3" };
 
-		data = 0;
+	data = 0;
 
-		for (int i = 0; i < 4; i++)
+	for (int i = 0; i < 4; i++)
+	{
+		if (m_io_regs[5] & 1 << i)
 		{
-			if (m_io_regs[5] & 1 << i)
-			{
-				data |= ioport(joynames[i])->read();
-				m_fd_data &= ~(1 << i);
-			}
+			data |= ioport(joynames[i])->read();
 		}
+	}
```

Comment:
On-the-fly tag lookups are error-prone and hurt performance.  Please use a `required_ioport_array<4>` data member.